### PR TITLE
#5512 - Implement post and query cases tests

### PR DESCRIPTION
**Pull request by BojteTamas**
- [X] Implemented the query endpoint
- [X] Implemented the status code checking

### CODE REVIEW

**Comment by BojteTamas** [comment]
image

File: sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     When API: I create a new case with an already created case
+     Then API: I check if the response contains TOO_OLD
+      And API: I check if the response has status 200
+  
+  Scenario: Create and check a case which is has invalid disease
+    Given API: I create a new person
+     When API: I create a new case with disease fun
+      And API: I check if the response has status 400
+     Then API: I check if the response contains Cannot deserialize value of type
+      And API: I check if the response contains not one of the values accepted for Enum
```
Comment (alinapopaiustina) [inline sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature:29]: Do we know what we expect in the body? What is the expected result here?

File: sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  
+  Scenario: Create and check successfully created Case by querying
+    Given API: I create a new person
+     When API: I create a new case
+      And API: I query the last created case
+      And API: I check if the response has status 200
+     Then API: I check if the response can be converted to a case object
```
Comment (alinapopaiustina) [inline sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature:36]: The test needs to be reliable: if this step fails, the test could qualify for false negative.

File: sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature [hunk 3]
```diff
@@ -0,0 +1,36 @@
+@Sanity @Person @API
+Feature: Check Person RestApi endpoints
+
+  Scenario: Create and check successfully created Case
+    Given API: I create a new person
+     When API: I create a new case
+     Then API: I check if the response contains OK
+      And API: I check if the response has status 200
```
Comment (alinapopaiustina) [inline sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature:8]: This test is redundant wit the last one.

**Review by alinapopaiustina — COMMENTED** [review]
Added some coments, thank you!

File: sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature (hunk 1 above)
Comment (BojteTamas) [reply sormas-e2e-tests/src/test/resources/features/sanity/api/Case.feature:29]: We are checking here the response body in order to see if it contains `Cannot deserialize value of type` or `not one of the values accepted for Enum`

**Comment by MateStrysewske** [comment]
@AndyBakcsy-she @RazvanHodisan Can you please check whether this PR is still relevant?

**Comment by rdutu-vg** [comment]
@MateStrysewske , closed, we already have this covered
